### ISSUE TITLE
How to make XTLS (V2ray) server for Iran on Ubuntu/Debian with the new Reality protocol.

### ISSUE THREAD

**Issue by SasukeFreestyle** [net4people/bbs#243]
Hello fellow Internet freedom-fighters.

I've been hearing many reports from my own users and others that v2ray servers are getting banned/blocked extremely fast.
This has also been the case for my own server as I've been changing IP almost every day to get around this.

I've been testing the new reality protocol and so far it seems to work great, no issues yet.
So I wrote another guide on how to make one for anyone interested.

https://github.com/SasukeFreestyle/XTLS-Iran-Reality

Note that reality does not support CDNs.

Keep the fight going!
Zan Zendegi Azadi
For freedom!

**Comment by prsgh** [comment]
In Advance special thanks for sharing your guide.
Do you face restrictions and handshake timeout on MTN irancell?

**Comment by MortThePersian** [comment]
Hello & Thank you so much for guiding others humbly

**Comment by SasukeFreestyle** [comment]
Hello!

You're welcome friend

I currently don't live in Iran, I'm an Iranian born outside of Iran and hosting a server for my relatives and friends that lives there and I'm hosting a server from my own home.

I've some users connected with Irancell and those users do not live in Tehran.
Some users (not all) can connect with Irancell in Tehran, some can't.

I've not yet tested different target websites in "dest" in the configuration and find one that works for everybody. So far I've only tested samsung.com.

My current understanding is that Irancell works great with CDN that has a clean IP, Amazon (AWS), Hetzner and Cloudflare have the best results. Some IPs of the CDN's are blocked/banned while others are not. Irancell blocks almost everything depending on where you live in Iran, city/region e.t.c

My relatives and friends report that Rightel works the best, but this is also not 100%.

ADSL / Fixed-line Internet / WIFI is 100% and works for everybody.

So in short, Try a different website ("dest") and ("servernames"). And if you find one that works 100% for Irancell let me know.

**Comment by MortThePersian** [comment]
> Hello!
> 
> You're welcome friend
> 
> I currently don't live in Iran, I'm an Iranian born outside of Iran and hosting a server for my relatives and friends that lives there and I'm hosting a server from my own home.
> 
> I've some users connected with Irancell and those users do not live in Tehran. Some users (not all) can connect with Irancell in Tehran, some can't.
> 
> I've not yet tested different target websites in "dest" in the configuration and find one that works for everybody. So far I've only tested samsung.com.
> 
> My current understanding is that Irancell works great with CDN that has a clean IP, Amazon (AWS), Hetzner and Cloudflare have the best results. Some IPs of the CDN's are blocked/banned while others are not. Irancell blocks almost everything depending on where you live in Iran, city/region e.t.c
> 
> My relatives and friends report that Rightel works the best, but this is also not 100%.
> 
> ADSL / Fixed-line Internet / WIFI is 100% and works for everybody.
> 
> So in short, Try a different website ("dest") and ("servernames"). And if you find one that works 100% for Irancell let me know.

Hello there;

I'm truly glad getting to know you, I also just made a quick explanation for those who are not familiar with the Iran situation.

https://github.com/XTLS/Xray-core/discussions/1961

Please continue your gracious movement..

I will also promise to do My best provide what I could


May The God's richest blessings shines upon the warriors who combat the darkness
